integration_test/gce/gce_testing.go

```diff
@@ -739,6 +739,95 @@ func runCommand(ctx context.Context, logger *log.Logger, stdin io.Reader, args [
 	return output, err
 }
 
+// getGcloudConfigDir returns the current gcloud configuration directory.
+func getGcloudConfigDir(ctx context.Context) (string, error) {
+	null, err := os.Open(os.DevNull)
+	if err != nil {
+		panic(fmt.Sprintf("unable to open os.DevNull: %v", err))
+	}
+	out, err := RunGcloud(ctx, log.New(null, "", 0), "", []string{"info", "--format=value[terminator=''](config.paths.global_config_dir)"})
+	if err != nil {
+		return "", fmt.Errorf("error running gcloud info: %w", err)
+	}
+	return out.Stdout, nil
+}
+
+// copyFile copies the contents of source into target.
+func copyFile(source string, target string) error {
+	in, err := os.Open(source)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+	out, err := os.Create(target)
+	if err != nil {
+		return err
+	}
+	_, copyErr := io.Copy(out, in)
+	syncErr := out.Sync()
+	closeErr := out.Close()
+	return multierr.Combine(copyErr, syncErr, closeErr)
+}
+
+// SetupGcloudConfigDir sets up a new gcloud configuration directory.
+// This copies the "configurations" subdirectory of the context-specified
+// configuration directory into the new directory.
+func SetupGcloudConfigDir(ctx context.Context, directory string) error {
+	relevantFiles := []string{
+		"active_config",
+		"access_tokens.db",
+		"credentials.db",
+		"default_configs.db",
```

Comment:
I worry that this list will go stale, since we are reaching into gcloud internals here. Why not just copy everything?